src/ecdh.rs

```diff
@@ -44,6 +47,31 @@ impl SharedSecret {
         }
     }
 
+    /// Creates a new shared secret from a pubkey and secret key with applied custom hash function
+    pub fn new_with_hash<F>(point: &PublicKey, scalar: &SecretKey, hash: &mut F) -> SharedSecret
+        where F: FnMut(&mut [u8], &[u8], &[u8]) -> i32
+    { 
+        extern "C" fn hash_callback<F>(output: *mut c_uchar, x: *const c_uchar, y: *const c_uchar, data: *const c_void) -> c_int 
+            where F: FnMut(&mut [u8], &[u8], &[u8]) -> i32
```

Comment:
I would consider actually using `i16` here. and casting it to `c_int` in the function. that way it should be compatible with any rust supporting architecture while being lossless.